hi-ml-histopathology/src/histopathology/datamodules/base_module.py

```diff
@@ -150,40 +185,126 @@ def _load_dataset(self, tiles_dataset: TilesDataset, stage: str, shuffle: bool)
+    fly by default. One can specify the tiling strategies (background removal, overlapping tiles, padding, ...) through
+    the class parameters.
+    """
+    def __init__(
+        self,
+        level: Optional[int] = 0,
+        tile_count: Optional[int] = None,
+        tile_size: Optional[int] = 224,
+        step: Optional[int] = None,
+        random_offset: Optional[bool] = True,
+        pad_full: Optional[bool] = False,
+        background_val: Optional[int] = 255,
+        filter_mode: Optional[str] = "min",
+        **kwargs: Any,
+    ) -> None:
+        """
+        :param level: the whole slide image level at which the image is extracted, defaults to 0
+        :param tile_count: number of tiles to extract, if None extracts all non-background tiles, defaults to None
+        :param tile_size: size of the square tile, defaults to 224
+        :param step: step size to create overlapping tiles, defaults to None
+        :param random_offset: randomize position of the grid, instead of starting from the top-left corner,
+        defaults to True
+        :param pad_full: pad image to the size evenly divisible by tile_size, defaults to False
+        :param background_val: the background constant to ignore background tiles (e.g. 255 for white background),
+        defaults to 255
+        :param filter_mode: mode must be in ["min", "max", "random"]. If total number of tiles is greater than
+        tile_count, then sort by intensity sum, and take the smallest (for min), largest (for max) or random (for
+        random) subset, defaults to "min" (which assumes background is high value)
+        """
+        super().__init__(**kwargs)
+        self.level = level
+        self.tile_count = tile_count
+        self.tile_size = tile_size
+        self.step = step
+        self.random_offset = random_offset
+        self.pad_full = pad_full
+        self.background_val = background_val
+        self.filter_mode = filter_mode
+
+    def _load_dataset(self, slides_dataset: SlidesDataset) -> Dataset:
+        base_transform = Compose(
+            [
+                LoadImaged(
+                    keys=slides_dataset.IMAGE_COLUMN,
+                    reader=WSIReader,
+                    backend="cucim",
+                    dtype=np.uint8,
+                    level=self.level,
+                    image_only=True,
+                ),
+                TileOnGridd(
+                    keys=slides_dataset.IMAGE_COLUMN,
+                    tile_count=self.tile_count,
+                    tile_size=self.tile_size,
+                    step=self.step,
+                    random_offset=self.random_offset,
+                    pad_full=self.pad_full,
+                    background_val=self.background_val,
+                    filter_mode=self.filter_mode,
+                    return_list_of_dicts=True,
+                ),
+            ]
```

Comment:
By default shouldn't we also end with a `ToTensorD` as in https://github.com/Project-MONAI/tutorials/blob/a810488d96b878f830473fb5b6498b035f010617/pathology/multiple_instance_learning/panda_mil_train_evaluate_pytorch_gpu.py#L369? What is the current shape of the dataloder output? this is also something we can try to address in the next PR when we integrate with deepmil, just a warning that this output will need to match what we have from the TilesDataModule

Reply:
The current output shape is (batch_size, tile_count, channels, img_size, img_size) Not sure if it matches the TilesDataModule but we can maybe address that in the next PR.